# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:8px;">**1. Introduction**</p>

**Author  : Salsa Sabitha Hurriyah**   

**Program ini dibuat untuk melakukan Data Exploration, Data Cleaning, dan Data Normalization.**

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:10px;">**2. Import Libraries**</p>

In [1]:
# Import libraries
import pandas as pd
import numpy as np

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:10px;">**3. Data Exploration**</p>

#### a. Data Loading

Membuat variabel `df` untuk menampung Data Frame dan membaca file dari format excel menggunakan `pd.read_csv()`

In [3]:
df = pd.read_csv('Financials.csv')

Menampilkan isi tabel:

In [3]:
df

Segment                   Country     Product   Discount Band   \
0          Government                    Canada   Carretera            None    
1          Government                   Germany   Carretera            None    
2           Midmarket                    France   Carretera            None    
3           Midmarket                   Germany   Carretera            None    
4           Midmarket                    Mexico   Carretera            None    
..                ...                       ...          ...             ...   
695    Small Business                    France    Amarilla            High    
696    Small Business                    Mexico    Amarilla            High    
697        Government                    Mexico     Montana            High    
698        Government                    Canada       Paseo            High    
699  Channel Partners  United States of America         VTT            High    

     Units Sold   Manufacturing Price   Sale Price     Gross Sales   \
0     $1,618.50                 $3.00       $20.00      $32,370.00    
1     $1,321.00                 $3.00       $20.00      $26,420.00    
2     $2,178.00                 $3.00       $15.00      $32,670.00    
3       $888.00                 $3.00       $15.00      $13,320.00    
4     $2,470.00                 $3.00       $15.00      $37,050.00    
..           ...                   ...          ...             ...   
695   $2,475.00               $260.00      $300.00    $7,42,500.00    
696     $546.00               $260.00      $300.00    $1,63,800.00    
697   $1,368.00                 $5.00        $7.00       $9,576.00    
698     $723.00                $10.00        $7.00       $5,061.00    
699   $1,806.00               $250.00       $12.00      $21,672.00    

         Discounts           Sales            COGS        Profit         Date  \
0             $-        $32,370.00      $16,185.00    $16,185.00   01/01/2014   
1             $-        $26,420.00      $13,210.00    $13,210.00   01/01/2014   
2             $-        $32,670.00      $21,780.00    $10,890.00   01/06/2014   
3             $-        $13,320.00       $8,880.00     $4,440.00   01/06/2014   
4             $-        $37,050.00      $24,700.00    $12,350.00   01/06/2014   
..              ...             ...             ...           ...         ...   
695   $1,11,375.00    $6,31,125.00    $6,18,750.00    $12,375.00   01/03/2014   
696     $24,570.00    $1,39,230.00    $1,36,500.00     $2,730.00   01/10/2014   
697      $1,436.40       $8,139.60       $6,840.00     $1,299.60   01/02/2014   
698        $759.15       $4,301.85       $3,615.00       $686.85   01/04/2014   
699      $3,250.80      $18,421.20       $5,418.00    $13,003.20   01/05/2014   

     Month Number  Month Name   Year  
0               1     January   2014  
1               1     January   2014  
2               6        June   2014  
3               6        June   2014  
4               6        June   2014  
..            ...          ...   ...  
695             3       March   2014  
696            10     October   2014  
697             2    February   2014  
698             4       April   2014  
699             5         May   2014  

[700 rows x 16 columns]

Dari data frame diatas dapat diketahui bahwa terdapat 700 baris dan 16 kolom

#### b. Tampilkan informasi summary
Untuk menampilkan informasi Summary dapat menggunakan method `.info()`
Method ini akan menampilkan informasi tabel berupa :  
nama-nama kolom, jumlah *missing value*, dan type data tiap kolom.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Segment                700 non-null    object
 1   Country                700 non-null    object
 2    Product               700 non-null    object
 3    Discount Band         700 non-null    object
 4    Units Sold            700 non-null    object
 5    Manufacturing Price   700 non-null    object
 6    Sale Price            700 non-null    object
 7    Gross Sales           700 non-null    object
 8    Discounts             700 non-null    object
 9     Sales                700 non-null    object
 10   COGS                  700 non-null    object
 11   Profit                700 non-null    object
 12  Date                   700 non-null    object
 13  Month Number           700 non-null    int64 
 14   Month Name            700 non-null    object
 15  Year                   

Dari tabel diatas dapat diketahui berapa banyak kolom dan apa saja nama kolom tersebut. Juga dapat diketahui *missing value* pada setiap kolom, beserta tipe data masing-masing kolom

#### c. Tampilkan list nama-nama kolom

Untuk menampilkan list nama-nama kolom pada data frame dapat menggunakan attribute `.columns`

In [5]:
df.columns

Index(['Segment', 'Country', ' Product ', ' Discount Band ', ' Units Sold ',
       ' Manufacturing Price ', ' Sale Price ', ' Gross Sales ', ' Discounts ',
       '  Sales ', ' COGS ', ' Profit ', 'Date', 'Month Number',
       ' Month Name ', 'Year'],
      dtype='object')

Diketahui bahwa ada beberapa nama kolom yang mengandung whitespace

#### d. Tampilkan list nilai unik masing-masing kolom `Segment`, `Country`, `Product`, `Discount Band`

Untuk menampilkan jumlah nilai unik pada tiap kolom dapat menggunakan method `.nunique()`

In [6]:
# Menampilkan berapa banyak nilai unique pada kolom `Segment`, `Country`, `Product`, `Discount Band`
# yang merupakan indeks kolom ke 0 hingga 4 (eksklusif)
df[df.columns[0:4]].nunique()

Segment            5
Country            5
 Product           6
 Discount Band     4
dtype: int64

Untuk menampilkan list nilai unik pada kolom dapat menggunakan method `.unique()`

In [7]:
# Menampilkan nilai unique pada kolom
for kolom in df.columns[0:4]:
    nilai_unik = df[kolom].unique()
    print(f"Nilai unik dari kolom {kolom}:")
    print(nilai_unik)
    print()

Nilai unik dari kolom Segment:
['Government' 'Midmarket' 'Channel Partners' 'Enterprise' 'Small Business']

Nilai unik dari kolom Country:
['Canada' 'Germany' 'France' 'Mexico' 'United States of America']

Nilai unik dari kolom  Product :
[' Carretera ' ' Montana ' ' Paseo ' ' Velo ' ' VTT ' ' Amarilla ']

Nilai unik dari kolom  Discount Band :
[' None ' ' Low ' ' Medium ' ' High ']



Diketahui terdapat whitespace pada value data frame

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:10px;">**4. Data Cleaning**</p>

#### a. Manipulasi kolom meliputi penggantian nama kolom, penghapusan kolom, atau penambahan kolom

##### - Menghapus kolom 'Month Number', 'Month Name', dan 'Year'

In [8]:
df.drop(columns = df.columns[13:16], inplace = True)
df.head()

Segment  Country     Product   Discount Band   Units Sold   \
0  Government   Canada   Carretera            None    $1,618.50    
1  Government  Germany   Carretera            None    $1,321.00    
2   Midmarket   France   Carretera            None    $2,178.00    
3   Midmarket  Germany   Carretera            None      $888.00    
4   Midmarket   Mexico   Carretera            None    $2,470.00    

   Manufacturing Price   Sale Price   Gross Sales   Discounts         Sales   \
0                $3.00       $20.00    $32,370.00        $-      $32,370.00    
1                $3.00       $20.00    $26,420.00        $-      $26,420.00    
2                $3.00       $15.00    $32,670.00        $-      $32,670.00    
3                $3.00       $15.00    $13,320.00        $-      $13,320.00    
4                $3.00       $15.00    $37,050.00        $-      $37,050.00    

          COGS        Profit         Date  
0   $16,185.00    $16,185.00   01/01/2014  
1   $13,210.00    $13,210.00   01/01/2014  
2   $21,780.00    $10,890.00   01/06/2014  
3    $8,880.00     $4,440.00   01/06/2014  
4   $24,700.00    $12,350.00   01/06/2014

Kolom yang  tidak digunakan atau tidak berpengaruh pada analisa akan dikeluarkan atau dihapus, pada data frame ini akan dihapus kolom 'Month Number', 'Month Name', dan 'Year'

##### - Mengganti nama kolom
Mengganti beberapa kolom yang memiliki whitespace pada nama kolom dan mengganti spasi dengan '_'

In [9]:
df = df.rename(columns={' Product ': 'Product', 
                   ' Discount Band ': 'Discount_Band',
                   ' Units Sold ': 'Units_Sold',
                   ' Manufacturing Price ': 'Manufacturing_Price',
                   ' Sale Price ': 'Sale_Price',
                   ' Gross Sales ': 'Gross_Sales',
                   ' Discounts ': 'Discounts',
                   '  Sales ': 'Sales',
                   ' COGS ': 'COGS',
                   ' Profit ': 'Profit'
                   })
df.columns

Index(['Segment', 'Country', 'Product', 'Discount_Band', 'Units_Sold',
       'Manufacturing_Price', 'Sale_Price', 'Gross_Sales', 'Discounts',
       'Sales', 'COGS', 'Profit', 'Date'],
      dtype='object')

Menghilangkan whitespace pada nama kolom dan mengganti spasi menjadi '_'

##### - Menghilangkan simbol '$' dan '()' pada kolom :  
'Units_Sold', 'Manufacturing_Price', 'Sale_Price', 'Gross_Sales', 'Discounts', 'Sales', 'COGS', 'Profit'

In [10]:
# Tabel tersebut berada pada index ke 4 sampai 11
df[df.columns[4:12]] = df[df.columns[4:12]].replace(to_replace = r'[ ()$,]', value = '', regex = True)
df.head()

Segment  Country      Product Discount_Band Units_Sold  \
0  Government   Canada   Carretera          None     1618.50   
1  Government  Germany   Carretera          None     1321.00   
2   Midmarket   France   Carretera          None     2178.00   
3   Midmarket  Germany   Carretera          None      888.00   
4   Midmarket   Mexico   Carretera          None     2470.00   

  Manufacturing_Price Sale_Price Gross_Sales Discounts     Sales      COGS  \
0                3.00      20.00    32370.00         -  32370.00  16185.00   
1                3.00      20.00    26420.00         -  26420.00  13210.00   
2                3.00      15.00    32670.00         -  32670.00  21780.00   
3                3.00      15.00    13320.00         -  13320.00   8880.00   
4                3.00      15.00    37050.00         -  37050.00  24700.00   

     Profit        Date  
0  16185.00  01/01/2014  
1  13210.00  01/01/2014  
2  10890.00  01/06/2014  
3   4440.00  01/06/2014  
4  12350.00  01/06/2014

Menghilangkan spasi dan simbol '$', '(', dan ')' pada value data frame agar kemudian kolom yang tipe datanya tidak sesuai dapat disesuaikan.

#### b. Check dan handling *missing value*

Pada data frame, list *missing value* tidak menggunakan yang standar, sehingga harus didefinisikan dan diubah menjadi list standar *missing value*

In [11]:
# Mendefinisikan list non-standard missing values yang mungkin terdapat dalam kolom
non_standard_missing_values = ['NA', 'N/A', 'null', '-']

# Mengganti non-standard missing values dengan standar yaitu NaN
df.replace(non_standard_missing_values, pd.NA, inplace=True)

# Menampilkan banyak missing values pada tiap kolom
df.isnull().sum()

Segment                 0
Country                 0
Product                 0
Discount_Band           0
Units_Sold              0
Manufacturing_Price     0
Sale_Price              0
Gross_Sales             0
Discounts              53
Sales                   0
COGS                    0
Profit                  5
Date                    0
dtype: int64

Mengganti *missing value* dengan angka 0, sehingga data dapat di proses

In [12]:
df = df_replaced2 = df.replace(to_replace=[np.nan], value = 0)
df

Segment                   Country      Product Discount_Band  \
0    Government                    Canada   Carretera          None    
1    Government                   Germany   Carretera          None    
2     Midmarket                    France   Carretera          None    
3     Midmarket                   Germany   Carretera          None    
4     Midmarket                    Mexico   Carretera          None    
..          ...                       ...          ...           ...   
675  Government  United States of America       Paseo          High    
676  Enterprise                   Germany       Paseo          High    
677   Midmarket                   Germany       Paseo          High    
678  Government  United States of America       Paseo          High    
679  Government                    Mexico       Paseo          High    

    Units_Sold Manufacturing_Price Sale_Price Gross_Sales  Discounts  \
0      1618.50                3.00      20.00    32370.00          0   
1      1321.00                3.00      20.00    26420.00          0   
2      2178.00                3.00      15.00    32670.00          0   
3       888.00                3.00      15.00    13320.00          0   
4      2470.00                3.00      15.00    37050.00          0   
..         ...                 ...        ...         ...        ...   
675     267.00               10.00      20.00     5340.00     801.00   
676    1085.00               10.00     125.00   135625.00   20343.75   
677    1175.00               10.00      15.00    17625.00    2643.75   
678    2007.00               10.00     350.00   702450.00  105367.50   
679    2151.00               10.00     350.00   752850.00  112927.50   

         Sales       COGS    Profit        Date  
0     32370.00   16185.00  16185.00  01/01/2014  
1     26420.00   13210.00  13210.00  01/01/2014  
2     32670.00   21780.00  10890.00  01/06/2014  
3     13320.00    8880.00   4440.00  01/06/2014  
4     37050.00   24700.00  12350.00  01/06/2014  
..         ...        ...       ...         ...  
675    4539.00    2670.00   1869.00  01/10/2013  
676  115281.25  130200.00  14918.75  01/10/2014  
677   14981.25   11750.00   3231.25  01/10/2014  
678  597082.50  521820.00  75262.50  01/11/2013  
679  639922.50  559260.00  80662.50  01/11/2013  

[680 rows x 13 columns]

In [13]:
# Check missing value
df.isnull().sum()

Segment                0
Country                0
Product                0
Discount_Band          0
Units_Sold             0
Manufacturing_Price    0
Sale_Price             0
Gross_Sales            0
Discounts              0
Sales                  0
COGS                   0
Profit                 0
Date                   0
dtype: int64

#### c. Penggantian tipe data yang sesuai dengan seharusnya.

In [14]:
# Mengganti tipe data pada kolom Manufacturing_Price, Sale_Price, Gross_Sales, Discounts, Sales, COGS, dan Profit
# (index kolom ke 5 sampai 11) menjadi float
df[df.columns[5:12]] = df[df.columns[5:12]].astype(float)

# Mengganti tipe data pada kolom Unit_Sold (index kolom ke 4) menjadi integer
df['Units_Sold'] = df['Units_Sold'].astype(float).astype(int)

# Mengganti tipe data pada kolom Date (index kolom ke 12) menjadi date time
df['Date'] = pd.to_datetime(df['Date'])


df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Segment              700 non-null    object        
 1   Country              700 non-null    object        
 2   Product              700 non-null    object        
 3   Discount_Band        700 non-null    object        
 4   Units_Sold           700 non-null    int32         
 5   Manufacturing_Price  700 non-null    float64       
 6   Sale_Price           700 non-null    float64       
 7   Gross_Sales          700 non-null    float64       
 8   Discounts            700 non-null    float64       
 9   Sales                700 non-null    float64       
 10  COGS                 700 non-null    float64       
 11  Profit               700 non-null    float64       
 12  Date                 700 non-null    datetime64[ns]
dtypes: datetime64[ns](1), float64(7), i

Penggantian tipe data dilakukan agar data dapat diproses selanjutnya. Pada data frame ini kolom yang tipe datanya akan diganti yaitu :  
- Manufacturing_Price, Sale_Price, Gross_Sales, Discounts, Sales, COGS, dan Profit dari object menjadi float  
- Units_Sold dari object menjadi integer
- Date dari object menjadi datetime

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:10px;">**5. Data Normalization**</p>


Lakukan normalisasi data yang sudah diolah sebelumnya sampai dengan 3NF.

- Membuat tabel Segment ID

In [36]:
segmentID = [f'S{i}' for i in range(len(df['Segment'].drop_duplicates()))]
segment_table = pd.DataFrame(data = {'Seg_ID' : segmentID,
                                    'Segment' : (df['Segment'].drop_duplicates()).copy()
                                    }
                            ).reset_index(drop=True)
segment_table.to_csv('Segment.csv', index=False)
segment_table

Seg_ID           Segment
0     S0        Government
1     S1         Midmarket
2     S2  Channel Partners
3     S3        Enterprise
4     S4    Small Business

- Membuat tabel Country ID

In [18]:
countryID = [f'C{i}' for i in range(len(df['Country'].drop_duplicates()))]
country_table  = pd.DataFrame(data = {'Country_ID' : countryID,
                                      'Country' : (df['Country'].drop_duplicates()).copy()
                                    }
                                ).reset_index(drop=True)
country_table.to_csv('Country.csv', index=False)
country_table

Country_ID                   Country
0         C0                    Canada
1         C1                   Germany
2         C2                    France
3         C3                    Mexico
4         C4  United States of America

- Membuat tabel Product ID

In [19]:
productID = [f'P{i}' for i in range(len(df['Product'].drop_duplicates()))]
product_table  = pd.DataFrame(data = {'Product_ID' : productID,
                                      'Product' : (df['Product'].drop_duplicates()).copy(),
                                      'Manufacturing_Price' : (df['Manufacturing_Price']).drop_duplicates().copy()
                                    }
                                ).reset_index(drop=True)
product_table.to_csv('Product.csv', index=False)
product_table

Product_ID      Product  Manufacturing_Price
0         P0   Carretera                   3.0
1         P1     Montana                   5.0
2         P2       Paseo                  10.0
3         P3        Velo                 120.0
4         P4         VTT                 250.0
5         P5    Amarilla                 260.0

- Membuat tabel discount_band ID

In [20]:
discountID = [f'D{i}' for i in range(len(df['Discount_Band'].drop_duplicates()))]
discount_table  = pd.DataFrame(data = {'Discount_ID' : discountID,
                                      'Discount_Band' : (df['Discount_Band'].drop_duplicates()).copy()
                                    }
                                ).reset_index(drop=True)
discount_table.to_csv('Discount.csv', index=False)
discount_table

Discount_ID Discount_Band
0          D0         None 
1          D1          Low 
2          D2       Medium 
3          D3         High

- Main Table

In [60]:
# Membuat tabel utama
mainTable = df[df.columns[0:13]]

# membuat series yang menyetor 
series = [f'P{i}' for i in range(len(mainTable))]

segmentID=[]
countryID=[]
productID = []
discountID = []

for i in range(len(mainTable)):
  for j in range(len(segment_table)):
    if mainTable.iloc[i,0] == segment_table.iloc[j,1]:
      segmentID.append(segment_table.iloc[j,0])
      break
  for j in range(len(country_table)):
    if mainTable.iloc[i,1] == country_table.iloc[j,1]:
      countryID.append(country_table.iloc[j,0])
      break
  for j in range(len(product_table)):
    if mainTable.iloc[i,2] == product_table.iloc[j,1]:
      productID.append(product_table.iloc[j,0])
      break
  for j in range(len(discount_table)):
    if mainTable.iloc[i,3] == discount_table.iloc[j,1]:
      discountID.append(discount_table.iloc[j,0])
      break

# Membuat table terpisah dari database tanpa referensi ke database
mainTable = mainTable.copy()

#Drop Kolom tidak berguna
mainTable.drop(columns = 'Segment', inplace = True)
mainTable.drop(columns = 'Country', inplace = True)
mainTable.drop(columns = 'Product', inplace = True)
mainTable.drop(columns = 'Discount_Band', inplace = True)

# Memasukkan kolom
mainTable.insert(0, column = 'SegmentID', value = segmentID)
mainTable.insert(1, column = 'CountryID', value = countryID)
mainTable.insert(2, column = 'ProductID', value = productID)
mainTable.insert(3, column = 'DiscountID', value = discountID)

mainTable.to_csv('MainTable.csv', index=False)

mainTable

SegmentID CountryID ProductID DiscountID  Units_Sold  Manufacturing_Price  \
0          S0        C0        P0         D0        1618                  3.0   
1          S0        C1        P0         D0        1321                  3.0   
2          S1        C2        P0         D0        2178                  3.0   
3          S1        C1        P0         D0         888                  3.0   
4          S1        C3        P0         D0        2470                  3.0   
..        ...       ...       ...        ...         ...                  ...   
695        S4        C2        P5         D3        2475                260.0   
696        S4        C3        P5         D3         546                260.0   
697        S0        C3        P1         D3        1368                  5.0   
698        S0        C0        P2         D3         723                 10.0   
699        S2        C4        P4         D3        1806                250.0   

     Sale_Price  Gross_Sales  Discounts      Sales      COGS    Profit  \
0          20.0      32370.0       0.00   32370.00   16185.0  16185.00   
1          20.0      26420.0       0.00   26420.00   13210.0  13210.00   
2          15.0      32670.0       0.00   32670.00   21780.0  10890.00   
3          15.0      13320.0       0.00   13320.00    8880.0   4440.00   
4          15.0      37050.0       0.00   37050.00   24700.0  12350.00   
..          ...          ...        ...        ...       ...       ...   
695       300.0     742500.0  111375.00  631125.00  618750.0  12375.00   
696       300.0     163800.0   24570.00  139230.00  136500.0   2730.00   
697         7.0       9576.0    1436.40    8139.60    6840.0   1299.60   
698         7.0       5061.0     759.15    4301.85    3615.0    686.85   
699        12.0      21672.0    3250.80   18421.20    5418.0  13003.20   

          Date  
0   2014-01-01  
1   2014-01-01  
2   2014-01-06  
3   2014-01-06  
4   2014-01-06  
..         ...  
695 2014-01-03  
696 2014-01-10  
697 2014-01-02  
698 2014-01-04  
699 2014-01-05  

[700 rows x 13 columns]

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:10px;">**6. Pengujian Database**</p>

#### 1. Setelah tabel sudah berhasil dibuat, uji keberhasilan database dan tabel yang kamu buat dengan kasus berikut:

##### a. Berikan tabel yang berisikan informasi total profit yang didapatkan di tiap jenis segmentasi. Jangan ambil data yang tidak diskon.

In [3]:
# Membaca file 5a.csv
df_5a = pd.read_csv('5_1_a.csv')
df_5a

segment  total_profit
0        Government   11388173.18
1    Small Business    4143168.50
2  Channel Partners    1316803.14
3        Enterprise     940096.88
4         Midmarket     660103.09

Berikut query sql nya:
``` sql
COPY (  
SELECT
	Segment.Segment,
	SUM(MainTable.Profit) AS total_profit
FROM MainTable  
JOIN Segment ON Segment.Seg_id = MainTable.SegmentID 
JOIN Discount ON Discount.Discount_ID = MainTable.DiscountID  
WHERE Discount.Discount_Band != 'None'  
GROUP BY Segment.Segment
ORDER BY 2 DESC
)  
TO 'C:\tmp\GC2\5_1_a.csv' WITH CSV 
DELIMITER ',' 
HEADER ;
```

##### b. Berikan tabel yang berisikan informasi rangkuman statistik yang memuat nilai rata-rata, min, dan max (dijadikan dalam kolom yang berbeda) dari Sales masing-masing negara.

In [4]:
# Membaca file 5a.csv
df_5b = pd.read_csv('5_1_b.csv')
df_5b

country  total_sales  min_sales  max_sales
0  United States of America  25029830.18    1730.54  1159200.0
1                    Canada  24887654.89    1655.08  1038082.5
2                    France  24354172.29    1763.86   962500.0
3                   Germany  23505340.82    1685.60  1017338.0
4                    Mexico  20949352.11    2508.66   848172.5

Berikut query sql nya :  

``` sql
COPY(  
SELECT
	Country.country,
	SUM(MainTable.sales) AS Total_sales,
	MIN(MainTable.sales) AS Min_sales,
	MAX(MainTable.sales) AS Max_sales
FROM MainTable  
JOIN Country ON Country.country_id = MainTable.CountryID
GROUP BY Country.country
ORDER BY 2 DESC
)  
TO 'C:\tmp\GC2\5_1_b.csv' WITH CSV 
DELIMITER ',' 
HEADER ;

#### 2. Simpan kedua tabel dalam file yang berbeda dan load ke dalam notebook pengolahan data untuk ditampilkan. Tuliskan query SQL pada markdown!

-> Jawaban pada nomor 5.1

# <p style="background-color:#FFC0CB;font-family:newtimeroman;color:#222021;font-size:130%;text-align:center;border-radius:10px 10px;margin-bottom:10px;">**7. Kesimpulan**</p>

1. Segment yang memiliki total profit tertinggi dengan data yang memiliki discount_band adalah Government dengan total 11388173.18  
  
2. Country (Negara) yang memiliki total sales tertinggi adalah United States of America	dengan total sales 25029830.18